src/pages/[platform]/build-a-backend/functions/examples/create_user_profile_record/index.mdx

```diff
@@ -0,0 +1,161 @@
+
+</Callout>
+
+You can use `defineAuth` and `defineFunction` to create a [Cognito post confirmation Lambda trigger](https://docs.aws.amazon.com/cognito/latest/developerguide/user-pool-lambda-post-confirmation.html) to create a profile record when a user is confirmed.
+
+<Callout info>
+
+A user is "confirmed" when they verify their account. Typically this happens when the user confirms their email via the verification email. The post confirmation handler will _not_ be triggered for federated sign-ins (i.e. social sign-in).
+
+</Callout>
+
+To get started, create a new directory and a resource file, `amplify/auth/post-confirmation/resource.ts`. Then, define the Function with `defineFunction`:
+
+```ts title="amplify/auth/post-confirmation/resource.ts"
+import { defineFunction } from '@aws-amplify/backend';
+
+export const postConfirmation = defineFunction({
+  entry: './post-confirmation-handler.ts',
+});
+```
+
+Then, create the corresponding handler file, `amplify/auth/post-confirmation/post-confirmation-handler.ts`, file with the following contents:
+
+```ts title="amplify/auth/post-confirmation/handler.ts"
+import type { PostConfirmationTriggerHandler } from "aws-lambda";
+import { Amplify } from "aws-amplify";
+import { generateClient } from "aws-amplify/data";
+import { type Schema } from "../../data/resource";
+import { env } from "$amplify/env/post-confirmation-handler";
+import { modelIntrospection } from "./amplifyconfiguration.json";
+
+Amplify.configure(
+  {
+    API: {
+      GraphQL: {
+        endpoint: env.AMPLIFY_DATA_GRAPHQL_ENDPOINT, 
+        region: env.AWS_REGION,
+        defaultAuthMode: "iam",
+        modelIntrospection: modelIntrospection as never,
+      },
+    },
+  },
+  {
+    Auth: {
+      credentialsProvider: {
+        getCredentialsAndIdentityId: async () => ({
+          credentials: {
+            accessKeyId: env.AWS_ACCESS_KEY_ID,
+            secretAccessKey: env.AWS_SECRET_ACCESS_KEY,
+            sessionToken: env.AWS_SESSION_TOKEN,
+          },
+        }),
+        clearCredentialsAndIdentityId: () => {
+          /* noop */
+        },
+      },
+    },
+  }
+);
+
+const client = generateClient<Schema>({
+  authMode: "iam",
+});
+
+export const handler: PostConfirmationTriggerHandler = async (event) => {
+  await client.models.UserProfile.create({
+    email: event.request.userAttributes.email,
+  });
+
+  return event;
+};
+
+
+```
+The `amplifyconfiguration.json` file is not automatically accessible to your function. To include it with your function, you must first do a deployment of your backend using either `npx amplify sandbox` or Amplify hosting. Then copy the `amplifyconfiguration.json` file that is placed in your project root into your function directory. Now you can import the file into your function as shown above.
```

Comment:
you can import the one that is generated in the project root and that'll get bundled, but this should be called out in a warning/info callout